---
title: "popgen_plots"
author: "Chandler Sutherland"
date: "2023-06-26"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(ggsignif)
library(ggpubr)
library(ggbeeswarm)
library(stats)
library(ggmosaic)
library(zoo)
require(tidyverse)
getwd()
```

```{r}
#read in col0_popgen_stats, generated by the Popgenome.Rmd script
col_popgen <- read_tsv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//col0_popgen_stats.tsv")

#add sample sizes to the label 
to_plot <- col_popgen %>% group_by(HV) %>%
  mutate(n=n()) %>%
  mutate(label = paste0(HV,"\n", "n=",n)) %>%
  mutate(PiNPiS = PiN/PiS) 

labels <- to_plot$label %>% unique()
to_plot$label <- factor(to_plot$label, levels=c(labels[1], labels[2]))
#to_plot$HV <- factor(to_plot$HV, levels=c('non-hv', 'hv'))

```


```{r}
p1 <- ggplot(to_plot,
       aes(x=label, y=D, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
  ylim(-3,3)+
   geom_signif(comparisons=list(c('non-hv\nn=98', 'hv\nn=36')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
    theme(legend.position='none', text=element_text(size=10))+
  geom_hline(yintercept=0, lwd=0.25)+
  ylab("Tajima's D")+
    theme(legend.position='none', text=element_text(size=10), axis.title.x=element_blank()) #+
  #geom_label(aes(label=name), position=position_quasirandom()) 
  
p1 

#ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\tajd.png', plot=p1, dpi='retina', width=2.25, height=1.5)

compare_means(D~HV, to_plot, method='wilcox.test', paired=FALSE)

to_plot %>% group_by(HV) %>% summarize(mean=mean(D), median=median(D))
```


```{r}
ylab=expression(omega)
p2 <- ggplot(to_plot,
       aes(x=label, y=R, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
  ylim(0,2)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(ylab)+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x=element_blank())
  
p2 
#ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pnps.png', plot=p2, dpi='retina', width=2.25, height=1.5)

compare_means(R~HV, to_plot, method='wilcox.test', paired=FALSE)
```

```{r}
pi_exp <- expression(pi)
p3 <- ggplot(to_plot,
       aes(x=label, y=Pi, fill=HV))+
    geom_violin(lwd=0.25, position = position_dodge(width = 0.9))+
  geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pi_exp)+
  #ylim(0, 0.09)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())
  
p3 
#ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\nucdiv.png', plot=p3, dpi='retina', width=2.2, height=1.5)
compare_means(Pi~HV, to_plot, method='wilcox.test', paired=FALSE)
```

For busted p, first step is a FDR correction, then plot as a mosaic
```{r}
#display a contigency table 
table <- table(col_popgen$HV, col_popgen$busted)
cont_table <- data.frame(no=c(0,7,7),yes=c(36,91,39+91),total=c(7+130+36+98), row.names=c("hv","non-hv", "total"))
cont_table2 <- data.frame(no=c(0,7),yes=c(36,91), row.names=c("hv","non-hv"))
```

```{r}
col_popgen$HV <- factor(col_popgen$HV, levels=c('non-hv', 'hv'))
col_popgen$busted <- factor(col_popgen$busted, levels=c('no', 'yes'))

p4 <- col_popgen %>%
    ggplot() +
    geom_mosaic(aes(x=product(HV), fill=busted, color=HV), offset=0.02) +
  geom_mosaic_text(aes(x = product(HV), fill = busted, label = after_stat(.wt)), size=2.5, color='black')+
  #theme_mosaic()+
    theme(axis.ticks.x = element_blank(), axis.ticks.y=element_blank())+
  ggtitle("Positive selection detected (FDR < 0.05)")+
 # theme(plot.title = element_text(size = 5))+
  theme(legend.position='none', 
        text=element_text(size=10), 
        axis.text.y = element_text(angle = 90,margin = margin(r = -10)),
        axis.text.x = element_text(angle = 0, margin=margin(r = -10)),
        axis.title=element_blank())+
    scale_fill_manual(values = c("white", "grey"))+
  scale_color_manual(values = c("hv" = '#F8766D',
                                "non-hv" = '#00BFC4'))

p4
#ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\busted.png', plot=p4, dpi='retina', width=2, height=1.5)


#to_plot$Gene %>% unique()%>% length()
```


```{r}
col_popgen %>%
    ggplot() +
    geom_mosaic(aes(x=product(HV), fill=busted), offset=0.02) +
  geom_mosaic_text(aes(x = product(HV), fill = busted, label = after_stat(.wt)), size=2.5)+
# theme_mosaic()+
 #   theme(axis.ticks.x = element_blank(), axis.ticks.y=element_blank())+
  #ggtitle("Positive selection detected (FDR < 0.05)")+
 # theme(plot.title = element_text(size = 5))+
#  ylab('Significant Positive Selection (BUSTED)')+
  theme(legend.position='none', 
        axis.ticks=element_blank(),
      #  text=element_text(size=10), 
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_text(angle = 90),
        #axis.text = element_text(vjust=-10), 
        plot.margin = margin(t = 0,  # Top margin
                             r = 0,  # Right margin
                             b = 0,  # Bottom margin
                             l = 0))+
    scale_fill_manual(values = c("white", "grey"))

ggplot(col_popgen, aes(fill=busted, y=busted, x=HV)) + 
    geom_bar(position="fill", stat="identity")

col_popgen$HV <- factor(col_popgen$HV, levels=c('non-hv', 'hv'))
col_popgen$busted <- factor(col_popgen$busted, levels=c('yes', 'no'))

ggplot(col_popgen, aes(x = HV, fill = busted, color=HV)) + 
  geom_bar(position = "fill", lwd=1.5) +
  labs(y = "Positive Selection Detected with BUSTED")+
  scale_fill_manual(values = c('no'="white", 'yes'="grey"))+
  scale_color_manual(values=c("hv" = '#F8766D',
                                "non-hv" = '#00BFC4'))+
  theme_classic()+
  scale_y_continuous(expand = c(0,0)) +
  annotate(geom="label", label='100%', x=2, y=0.5, color='#F8766D', size=4, fill='white')+
  annotate(geom="label", label='93%', x=1, y=0.5, color='#00BFC4', size=4, fill='white')+
  annotate(geom="label", label='7%', x=1, y=0.035, color='#00BFC4', size=4, fill='white')+
  annotate(geom='text', label='yes', x=0.5, y=0.5, angle=90)+
  annotate(geom='text', label='no', x=0.5, y=0.025, angle=90)+
  theme(axis.text.y=element_blank(), 
        axis.text.x=element_text(margin=margin(r=-30)), 
        line=element_blank(), 
        axis.title.x=element_blank(), 
        axis.title.y=element_text(margin=margin(r=0)))+
  guides(color='none')
```

```{r}
fisher.test(table)
```

```{r}

pis_div_pin=expression(pi['n']~'/'~pi['s'])
ggplot(to_plot, 
         aes(x=label, y=PiNPiS, fill=label))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test)+
  ylim(0,2.5)+
 # ylim(0,2)+ 'non-hv\nn=98', 'hv\nn=36'
    scale_fill_manual(values = c('hv\nn=36' = "#F8766D",
                               'non-hv\nn=98' = '#00BFC4'))+
  theme_classic()+
  labs(x = '', y=pis_div_pin)+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10))

ggplot(pn_ps_conservative, 
         aes(x=label, y=PiNPiS, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test)+
 # ylim(0,2)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  labs(x = '', y=pis_div_pin)+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10))

compare_means(PiNPiS~HV, to_plot, method='wilcox.test', paired=FALSE)
```


```{r}
pis=expression(pi['S'])
pin=expression(pi['N'])
pis_equals_pin=expression(pi['s']~'='~pi['n'])
#paml %>% filter(dnds_ML > 1) %>% subset(select=c('Gene', 'Clade', 'name', 'HV', 'R', 'dnds_ML'))

pis_piN <- to_plot %>% ungroup() %>% mutate(clean_name = case_when(name == 'RPP13' ~ 'RPP13', 
                                                        name == 'RPP7' ~ 'RPP7', 
                                                        name == 'DSC1' ~ 'DSC1', 
                                                        name == 'RPP8' ~ 'RPP8', 
                                                        name == 'RPS5' ~ 'RPS5')) %>%
  ggplot(aes(x=PiS, y=PiN, fill=HV))+
    geom_point(aes(color=HV), size=0.5)+
  geom_abline(slope=1, intercept=0, color='darkgrey', linetype='dashed')+
  ylim(0,0.08)+
  xlim(0,0.08)+
  scale_fill_manual(values = c("hv" = "#F8766D",
                               "non-hv" = '#00BFC4'))+
  geom_text(aes(label=clean_name), size=3)+
  xlab(pis)+
  ylab(pin)+
  theme_classic()+
  theme(text=element_text(size=11), 
        legend.text=element_text(size=11), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))+
  annotate(geom="text", label=pis_equals_pin, x=0.06, y=0.0625, angle=40, vjust=-1, color='darkgrey', size=3)


ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pis_pin.png', plot=pis_piN, dpi='retina', width=5.4, height=3.6)

```

```{r}
pis_plot <- ggplot(to_plot,
       aes(x=label, y=PiS, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
 # ylim(0,2)+
  ylim(0, 0.1)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pis)+
 # geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x=element_blank())
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pis.png', plot=pis_plot, dpi='retina', width=2, height=2.5)

pin_plot <- ggplot(to_plot,
       aes(x=label, y=PiN, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
 # ylim(0,0.1)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pin)+
 # geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x=element_blank())
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pin.png', plot=pin_plot, dpi='retina', width=2, height=2.5)
```


```{r}

pi.labs <- c(pin, pis)
names(pi.labs) <- c("PiN", "PiS")
p13 <- to_plot %>% subset(select=c('name', 'PiS', 'PiN', 'HV')) %>% pivot_longer(
    cols = c(PiS, PiN), 
    names_to = "pi_type",
    values_to = "value") %>%
  ggplot(aes(x=HV, y=value, fill=HV))+
  geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
  ylim(0,0.1)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
 # geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())+
  facet_wrap(~pi_type, labeller=labeller(pi_type=pi.labs))

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pisandpin.png', plot=p13, dpi='retina', width=2.2, height=1.75)

```


```{r}
```

```{r}
ylab=expression(P[n]/P[s])
ggplot(to_plot,
       aes(x=label, y=R, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
  ylim(0,2)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())+
  facet_wrap(~busted)+
  geom_label(aes(label=name), position=position_quasirandom())
```


```{r}
to_plot %>% filter(busted == 'no' & R > 1)
#2 > dS > 0.02, and dN <2
to_plot %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2) %>% ggplot(
       aes(x=label, y=R, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
  ylim(0,2)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())+
  facet_wrap(~busted)+
  geom_label(aes(label=name), position=position_quasirandom()) 

pn_ps_conservative <- to_plot %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2) %>% group_by(HV) %>%
  mutate(n=n()) %>%
  mutate(label = paste0(HV,"\n", "n=",n))

labels <- pn_ps_conservative$label %>% unique()
pn_ps_conservative$label <- factor(pn_ps_conservative$label, levels=c(labels[1], labels[2]))

ggplot(pn_ps_conservative,
       aes(x=label, y=R, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
 # ylim(0,2)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())

ggplot(pn_ps_conservative,
       aes(x=label, y=R, fill=HV))+
    geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c(labels[1], labels[2])), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
 # ylim(0,2)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  geom_hline(yintercept=1, lwd=0.25)+
    theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())+
  facet_wrap(~busted)
```

```{r}
 ggplot(to_plot, aes(x=Pi, y=D, fill=HV))+
    geom_point(aes(color=HV))+
#  ylim(0, 0.15)+
#  xlim(0,0.15)+
  scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  geom_text(aes(label=name), size=3)+
  xlab(pi_exp)+
  ylab("Tajima's D")+
  theme_classic() #+
  #theme(text=element_text(size=11), 
  #      legend.text=element_text(size=11), 
  #      legend.title = element_blank(), 
   #     legend.key.size = unit(.2, 'cm'), 
    #    legend.key.height=unit(.2, 'cm'),
     #   legend.key.width = unit(.2, 'cm'))

```

```{r}
col_popgen %>% filter(name=='RPS5')
```

```{r}
#read in col0_popgen_stats, generated by the Popgenome.Rmd script
col_per_domain <- read_tsv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//col0_per_domain_stats.tsv")%>% mutate(PiNPiS = PiN/PiS)

col_per_domain[c('PiNPiS')][sapply(col_per_domain[c('PiNPiS')], is.infinite)] <- NA

cds <- to_plot %>% subset(select=c('Clade', 'Gene', 'HV', 'name', 'D', 'Pi', 'PiN', 'PiS', 'PiNPiS')) %>% mutate('domain'='CDS')
all <- rbind(col_per_domain %>% subset(select=-c(start, stop)), cds)

```
```{r}
all <- all %>%
  mutate(domain=recode(domain, 'cc' = "CC", 'tir'="TIR", 'nbarc'='NBARC', 'lrr'='LRR'))

all$domain <- factor(all$domain, levels=c('CDS', 'CC', 'TIR', 'NBARC', 'LRR'))
p7 <- ggplot(all,
       aes(x=domain, y=Pi, fill=HV))+
  geom_boxplot(lwd=0.25, outlier.size=0.1)+
    #geom_violin(lwd=0.25, outlier.size = 0.1, alpha=0.1)+
  #geom_point(aes(fill=HV, alpha=0.1), size=0.1, position=position_jitterdodge())+
  #geom_quasirandom(aes(color=HV), size=0.1)+
   #geom_signif(stat = "identity",
    #          data = data.frame(x = c(0.7, 1.7, 2.7, 3.7, 4.7),
     #                           xend = c(1.3, 2.3, 3.3, 4.3, 5.3),
      #                          y = c(0.2, 0.3, 0.4, 0.3, 0.4),
       #                         annotation = c("***", " *** ", "  ***  ", '   ***   ', '     ***     ')),
        #      aes(x = x,
         #         xend = xend,
          #        y = y,
           #       yend = y,
            #      annotation = annotation, 
             #  textsize=2))+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pi_exp)+
  #ylim(0, 0.5)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())

  
p7 
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pi_domain.png', plot=p7, dpi='retina', width=2.2, height=1.5)

compare_means(Pi~HV, all %>% filter(domain=='CDS'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='NBARC'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='CC'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='TIR'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='LRR'), method='wilcox.test', paired=FALSE)$p/5

compare_means(Pi~domain, all, method='wilcox.test')
compare_means(Pi~domain, all %>% filter(HV=='hv'), method='wilcox.test')
compare_means(Pi~domain, all %>% filter(HV=='non-hv'), method='wilcox.test')

kruskal.test(Pi ~ domain, data = all)
kruskal.test(Pi ~ domain, data = all%>%filter(HV=='hv'))
kruskal.test(Pi ~ domain, data = all%>%filter(HV=='non-hv'))
```

```{r}
all %>% filter(Clade == 'Int9687_297_427_R_29')
to_plot %>% filter(Clade == 'Int9687_297_427_R_29')
all
```


```{r}
p10 <- ggplot(all,
       aes(x=domain, y=D))+
    #geom_violin(lwd=0.25, aes(fill=HV))+
   geom_boxplot(lwd=0.25, aes(fill=HV), outlier.size = 0.1)+
 # geom_quasirandom(aes(fill=HV, alpha=0.1), size=0.1, dodge.width=.9)+
  # geom_signif(stat = "identity",
   #           data = data.frame(x = c(0.7, 1.7, 2.7, 3.7),
    #                            xend = c(1.3, 2.3, 3.3, 4.3),
     #                           y = c(1.75, 4, 3, 4),
      #                          annotation = c("***", " *** ", "  ***  ", '   ***   ')),
       #       aes(x = x,
        #          xend = xend,
         #         y = y,
          #        yend = y,
           #       annotation = annotation, 
            #   textsize=2))+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab("Tajima's D")+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())

  
p10 
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_domain.png', plot=p10, dpi='retina', width=2.2, height=1.5)

compare_means(D~HV, all %>% filter(domain=='NBARC'), method='wilcox.test', paired=FALSE)$p/4
compare_means(D~HV, all %>% filter(domain=='CC'), method='wilcox.test', paired=FALSE)$p/4
compare_means(D~HV, all %>% filter(domain=='TIR'), method='wilcox.test', paired=FALSE)$p/4
compare_means(D~HV, all %>% filter(domain=='LRR'), method='wilcox.test', paired=FALSE)$p/4

compare_means(D~domain, all %>% filter(HV=='hv'), method='wilcox.test')
compare_means(D~domain, all %>% filter(HV=='non-hv'), method='wilcox.test')

kruskal.test(D ~ domain, data = all)
kruskal.test(D ~ domain, data = all%>%filter(HV=='hv'))
kruskal.test(D ~ domain, data = all%>%filter(HV=='non-hv'))
```

```{r}
ggplot(all,
       aes(x=domain, y=PiNPiS, fill=HV))+
    geom_violin(lwd=0.25)+
  geom_quasirandom(alpha=0.3, size=0.5, dodge.width=.9)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pin)+
  #ylim(0, 0.22)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())

compare_means(PiN~domain, col_per_domain %>% filter(HV=='hv'), method='wilcox.test')
compare_means(PiN~domain, col_per_domain %>% filter(HV=='non-hv'), method='wilcox.test')
```


```{r}
ggplot(col_per_domain,
       aes(x=domain, y=PiS, fill=HV))+
    geom_violin(lwd=0.25, position = position_dodge(width = 0.9))+
  geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE, dodge.width = 0.9)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pis)+
  ylim(0, 0.22)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())

compare_means(PiS~domain, col_per_domain %>% filter(HV=='hv'), method='wilcox.test')
compare_means(PiS~domain, col_per_domain %>% filter(HV=='non-hv'), method='wilcox.test')
```


```{r}
#cds <- to_plot %>%  filter(PiS > 0.02 & PiS < 2 & PiN < 2) %>% subset(select=c('Gene', 'Clade', 'PiNPiS', 'HV')) %>% mutate('domain'='CDS')

#domains <- col_per_domain %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2) %>% subset(select=c('Gene', 'Clade', 'PiNPiS', 'domain', 'HV'))

pinpis_plot <- all %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2)

p11 <- ggplot(all,
       aes(x=domain, y=PiNPiS, fill=HV))+
    geom_boxplot(lwd=0.25, outlier.size=0.1)+
  #geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE, dodge.width = 0.9)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pis_div_pin)+
  ylim(0, 1)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pinpis_domain.png', plot=p11, dpi='retina', width=2.2, height=1.5)

compare_means(PiNPiS~HV, pinpis_plot %>% filter(domain=='NB-ARC'), method='wilcox.test', paired=FALSE)$p/3
compare_means(PiNPiS~HV, pinpis_plot %>% filter(domain == 'LRR'), method='wilcox.test', paired=FALSE)$p/3
compare_means(PiNPiS~HV, pinpis_plot %>% filter(domain=='CDS'), method='wilcox.test', paired=FALSE)$p/3

compare_means(PiNPiS~domain, col_per_domain %>% filter(HV=='hv') %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2), method='wilcox.test')
compare_means(PiNPiS~domain, col_per_domain %>% filter(HV=='non-hv') %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2), method='wilcox.test')

kruskal.test(PiNPiS ~ domain, data = pinpis_plot)
kruskal.test(PiNPiS ~ domain, data = pinpis_plot%>%filter(HV=='hv'))
kruskal.test(PiNPiS ~ domain, data = pinpis_plot%>%filter(HV=='non-hv'))
```

```{r}
ggplot(all,
       aes(x=domain, y=PiNPiS, fill=HV))+
    geom_violin(lwd=0.25, position = position_dodge(width = 0.9))+
  geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE, dodge.width=0.9)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab('Tajima D')+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())
```


```{r}
ggplot(col_per_domain,
       aes(x=HV, y=PiN, fill=HV))+
    geom_violin(lwd=0.25, position = position_dodge(width = 0.9))+
  geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab(pin)+
#  ylim(0, 0.22)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())+
  facet_wrap(~domain)

col_per_domain %>% filter(PiS > 0.02 & PiS < 2 & PiN < 2) %>%
ggplot(aes(x=HV, y=PiNPiS, fill=HV))+
    geom_violin(lwd=0.25, position = position_dodge(width = 0.9))+
  geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab('PiN/PiS')+
#  ylim(0, 0.22)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())+
  facet_wrap(~domain)

ggplot(col_per_domain,
       aes(x=HV, y=D, fill=HV))+
    geom_violin(lwd=0.25, position = position_dodge(width = 0.9))+
  geom_quasirandom(alpha=0.3, size=0.5, method = 'quasirandom', varwidth = TRUE)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
  ylab('Tajima D')+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())+
  facet_wrap(~domain)
```

```{r}
col_per_domain %>% filter(name=='RPS4')
col_per_domain
```
```{r}
col_slider <- read_csv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//pi_slider_50.csv")
col_slider2 <- read_csv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//pi_slider_100.csv")
col_slider2 <- col_slider2 %>% mutate(nuc_midpoint = (start+50)*3) %>% mutate(nuc_length=length*3) %>% mutate(normal=nuc_midpoint/nuc_length) %>% rename(Clade=clade)

col_slider <- col_slider %>% mutate(normal=start/length*100) %>% rename(Clade=clade) %>% subset(select=-`...1`)
title <- col_per_domain %>% subset(select=c('Clade', 'Gene', 'HV', 'name')) %>% unique()
named <- col_slider %>% merge(title, all.y=TRUE)
named2 <- col_slider2 %>% merge(title, all.y=TRUE)


```
```{r}
p8 <- ggplot(data=named2, aes(x=normal, y=Pi, color=HV))+
 # geom_line(aes(group=Clade), size=0.5,alpha=0.1)+
  geom_smooth(lwd=0.5)+
  theme_classic()+
  xlab('Normalized Nucleotide Position')+
  ylab(pi_exp)+
 # theme(legend.position='right', text=element_text(size=10))+
    theme(text=element_text(size=10), 
        legend.text=element_text(size=10), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))

p8 
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pi_window.png', plot=p8, dpi='retina', width=3.2, height=1.5)

p9 <- ggplot(data=named2, aes(x=normal, y=D, color=HV))+
 # geom_line(aes(group=Clade), size=0.5,alpha=0.1)+
  geom_smooth(lwd=0.5)+
  theme_classic()+
  xlab('Normalized Nucleotide Position')+
  ylab("Tajima's D")+
 # ylim(-1.5, 1.5)+
 # theme(legend.position='right', text=element_text(size=10))+
    theme(text=element_text(size=10), 
        legend.text=element_text(size=10), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))

p9 
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_window.png', plot=p9, dpi='retina', width=3.2, height=1.5)
```


```{r}
test <- named2 %>% filter(name %in% c('RPP13'))
test %>%
  group_by(Clade)%>%
  mutate(moving_avg = rollmean(Pi, k=10, fill=NA, align='right', na.pad=T))

ggplot(test, aes(x=normal, y=Pi, color=HV))+
  geom_point()+
  theme_classic()

ggplot(data = test, aes(x = normal, y = Pi, group=HV)) + 
   stat_summary(geom = "line", fun.y = mean)

ggplot(data=named, aes(x=normal, y=PiN, color=HV))+
  #geom_point(size=0.5,alpha=0.5)+
  geom_smooth()+
  theme_classic()

ggplot(data=named, aes(x=normal, y=PiS, color=HV))+
  #geom_point(size=0.5,alpha=0.5)+
  geom_smooth()+
  theme_classic()



ggplot(data=named, aes(x=normal, y=D, color=HV))+
  geom_point(size=0.5,alpha=0.1)+
  geom_smooth()+
  theme_classic()
```

```{r}
ggplot(data=named2, aes(x=normal, y=D, color=HV))+
 geom_point(aes(group=Clade, alpha=0.01))+
  geom_smooth(se=FALSE)+
  theme_classic()
```
```{r}
named %>% group_by(Clade, HV) %>% summarize(na_piss=sum(is.na(PiS))) %>% arrange(na_piss)

named %>% filter(name=='RPP13') %>%
  ggplot(aes(x=start, y=Pi, color=HV))+
 geom_line(aes(group=Clade, alpha=0.01))+
  geom_smooth(se=FALSE)+
  theme_classic()
```
```{r}
named2 %>% filter(Gene=='AT5G43725' | Gene == 'AT5G43740') %>% ggplot(aes(x=nuc_midpoint, y=Pi, color=HV))+
  geom_point()+
  theme_classic()

named %>% filter(Gene == 'AT5G43725')

named2 %>% filter(name=='RPP7') %>% ggplot(aes(x=nuc_midpoint, y=Pi, color=HV))+
  geom_line()+
  theme_classic()
```
non-hvNLR AT5G43725 and hvNLR AT5G43740

```{r}
col_per_domain %>% ungroup() %>% mutate(clean_name = case_when(name == 'RPP13' ~ 'RPP13', 
                                                        name == 'RPP7' ~ 'RPP7', 
                                                        name == 'RPP1' ~ 'RPP1', 
                                                        name == 'RPP8' ~ 'RPP8', 
                                                        name == 'RPS5' ~ 'RPS5')) %>%
  ggplot(aes(x=PiS, y=PiN, fill=HV))+
    geom_point(aes(color=HV), size=0.5)+
  geom_abline(slope=1, intercept=0, color='darkgrey', linetype='dashed')+
  ylim(0,0.08)+
  xlim(0,0.08)+
  scale_fill_manual(values = c("hv" = "#F8766D",
                               "non-hv" = '#00BFC4'))+
  geom_text(aes(label=clean_name), size=3)+
  xlab(pis)+
  ylab(pin)+
  theme_classic()+
  theme(text=element_text(size=11), 
        legend.text=element_text(size=11), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))+
  annotate(geom="text", label=pis_equals_pin, x=0.06, y=0.0625, angle=40, vjust=-1, color='darkgrey', size=3)+
  facet_wrap(~domain)
```


```{r}
named %>% subset(select=c('Gene','name', 'PiS', 'PiN', 'HV', 'normal')) %>% pivot_longer(
    cols = c(PiS, PiN), 
    names_to = "pi_type",
    values_to = "value")%>%
  ggplot(aes(x=normal, y=value, color=pi_type))+
  geom_smooth(se=FALSE)+
  theme_classic()+
  facet_wrap(~HV)
  
```

```{r}
named %>% mutate(PiNPiS = PiN/PiS) %>% ggplot(aes(x=normal, y=PiNPiS, color=HV))+
 # geom_point(size=0.5,alpha=0.1)+
  geom_smooth()+
  theme_classic()
```

```{r}
col_per_domain <- col_per_domain %>%
  mutate(domain=recode(domain, 'cc' = "CC", 'tir'="TIR", 'nbarc'='NBARC', 'lrr'='LRR'))

col_per_domain$domain <- factor(col_per_domain$domain, levels=c('CC', 'TIR', 'NBARC', 'LRR'))

named2 %>% filter(name=='RPP13') %>% ggplot() + 
  geom_rect(data=col_per_domain %>% filter(name=='RPP13'), aes(xmin=start*3, xmax=stop*3,
                              ymin=Pi, ymax=Pi, color=domain),size=5)+
  geom_point(aes(x=nuc_midpoint, y=Pi))+
  geom_line(aes(x=nuc_midpoint, y=Pi))+
  xlim(0, 3351)+
  theme_classic()
  #geom_abline(slope=0, intercept=0, color='darkgrey', linetype='dashed')

col_per_domain %>% filter(name=='RPP8')
```
"AT1G31540" "AT1G59780" "AT1G62630" "AT1G69550" "AT2G14080" "AT4G11170"
```{r}
gene='AT1G58602'
#col_per_domain$domain <- 
named2 %>% filter(Gene==gene) %>% ggplot() + 
    geom_point(aes(x=nuc_midpoint, y=Pi))+
    geom_line(aes(x=nuc_midpoint, y=Pi))+
  geom_rect(data=col_per_domain %>% filter(Gene==gene), aes(xmin=start*3, xmax=stop*3,
                             ymin=Pi, ymax=Pi, color=domain),size=5)+
  ylim(0, 0.1)+
 # geom_line(aes(x=nuc_midpoint, y=Pi))+
  ggtitle(gene)+
 # xlim(0, 3351)+
  theme_classic()
  #ylim(0, 0.1)

filter(all, Gene==gene)


```

```{r}
gene='AT4G26090'
#col_per_domain$domain <- 
named2 %>% filter(Gene==gene) %>% ggplot() + 
    geom_point(aes(x=nuc_midpoint, y=D))+
    geom_line(aes(x=nuc_midpoint, y=D))+
  geom_rect(data=col_per_domain %>% filter(Gene==gene), aes(xmin=start*3, xmax=stop*3,
                             ymin=D, ymax=D, color=domain),size=5)+
  #ylim(0, 0.1)+
 # geom_line(aes(x=nuc_midpoint, y=Pi))+
  ggtitle(gene)+
 # xlim(0, 3351)+
  theme_classic()

all %>% filter(Gene=='AT4G16960')
```

non-hvNLR 
AT5G43730 and hvNLR AT5G43740
```{r}
gene='AT5G43730'
#col_per_domain$domain <- 
Pi_AT5G43730 <- named2 %>% filter(Gene=='AT5G43730') %>% ggplot() + 
  geom_rect(data=col_per_domain %>% filter(Gene=='AT5G43730'), aes(xmin=start*3, xmax=stop*3,
                             ymin=0, ymax=0, color=domain),size=5)+
   scale_color_manual(values=c('#00997F', '#00BADE', '#9661FF'))+
    geom_point(aes(x=nuc_midpoint, y=Pi))+
    geom_line(aes(x=nuc_midpoint, y=Pi))+
  
  ylim(0, 0.1)+
 # geom_line(aes(x=nuc_midpoint, y=Pi))+
  ggtitle('AT5G43730')+
  xlab('300bp Window Midpoint')+
  ylab(pi_exp)+
 # xlim(0, 3351)+
  theme_classic()+
  theme(legend.position='none')
  #ylim(0, 0.1)

gene='AT5G43740'
#col_per_domain$domain <- 
Pi_AT5G43740 <- named2 %>% filter(Gene=='AT5G43740') %>% ggplot() + 
  geom_rect(data=col_per_domain %>% filter(Gene=='AT5G43740'), aes(xmin=start*3, xmax=stop*3,
                             ymin=D, ymax=0, color=domain),size=5)+
   scale_color_manual(values=c('#00997F', '#00BADE', '#9661FF'))+
    geom_point(aes(x=nuc_midpoint, y=Pi))+
    geom_line(aes(x=nuc_midpoint, y=Pi))+
  
  ylim(0, 0.1)+
 # geom_line(aes(x=nuc_midpoint, y=Pi))+
  ggtitle('AT5G43740')+
  xlab('300bp Window Midpoint')+
  ylab(pi_exp)+
 # xlim(0, 3351)+
  theme_classic()+
  theme(legend.position='none')

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\Pi_case.png', plot=Pi_AT5G43740, dpi='retina', width=5, height=3)

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\Pi_case2.png', plot=Pi_AT5G43730, dpi='retina', width=5, height=3)
```


```{r}
gene='AT5G43730'
#col_per_domain$domain <- 
D_AT5G43730 <- named2 %>% filter(Gene=='AT5G43730') %>% ggplot() + 
  geom_rect(data=col_per_domain %>% filter(Gene=='AT5G43730'), aes(xmin=start*3, xmax=stop*3,
                             ymin=D, ymax=D, color=domain),size=5)+
   scale_color_manual(values=c('#00997F', '#00BADE', '#9661FF'))+
    geom_point(aes(x=nuc_midpoint, y=D))+
    geom_line(aes(x=nuc_midpoint, y=D))+
  
  ylim(-2, 2)+
 # geom_line(aes(x=nuc_midpoint, y=Pi))+
  ggtitle('AT5G43730')+
  xlab('300bp Window Midpoint')+
  ylab("Tajima's D")+
 # xlim(0, 3351)+
  theme_classic()+
  theme(legend.position='none')
  #ylim(0, 0.1)

gene='AT5G43740'
#col_per_domain$domain <- 
D_AT5G43740 <- named2 %>% filter(Gene=='AT5G43740') %>% ggplot() + 
  geom_rect(data=col_per_domain %>% filter(Gene=='AT5G43740'), aes(xmin=start*3, xmax=stop*3,
                             ymin=D, ymax=D, color=domain),size=5)+
   scale_color_manual(values=c('#00997F', '#00BADE', '#9661FF'))+
    geom_point(aes(x=nuc_midpoint, y=D))+
    geom_line(aes(x=nuc_midpoint, y=D))+
  
  ylim(-2, 2)+
 # geom_line(aes(x=nuc_midpoint, y=Pi))+
  ggtitle('AT5G43740')+
  xlab('300bp Window Midpoint')+
  ylab("Tajima's D")+
 # xlim(0, 3351)+
  theme_classic()+
  theme(legend.position='none')

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_case.png', plot=D_AT5G43740, dpi='retina', width=5, height=3)

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_case2.png', plot=D_AT5G43730, dpi='retina', width=5, height=3)
```